src/Xamarin.Android.Build.Tasks/Tasks/ConvertResourcesCases.cs

```diff
@@ -94,9 +92,11 @@ void FixupResources (ITaskItem item)
 					resourcedirectories, (level, message) => {
 						switch (level) {
 						case TraceLevel.Error:
+							resource_name_case_map ??= MonoAndroidHelper.LoadResourceCaseMap (ResourceNameCaseMap);
```

Comment:
I wonder if it would be "easier to maintain" if we instead turned the `resource_name_case_map` field into a *property*:

```csharp
Dictionary<string, string>? _resource_name_case_map;
Dictionary<string, string> resource_name_case_map =>
    _resource_name_case_map ??= MonoAndroidHelper.LoadResourceCaseMap (ResourceNameCaseMap);
```

This would reduce (remove?) the possibility of `NullReferenceException`s because `resource_name_case_map` was used before it was initialized.

Reply:
I made this change, and also used `RegisterTaskObject` so we no longer create a giant string at all.

Instead of the `$(_AndroidResourceNameCaseMap)` property, we can just use the same `Dictionary<string,string>` throughout the build.

This gave us another 50ms or so.